components/page-templates/B2B/CreateNewQuoteTemplate/CreateNewQuoteTemplate.spec.tsx

```diff
@@ -0,0 +1,239 @@
+      </button>
+    </div>
+  ),
+}))
+
+const addToQuoteTest = async () => {
+  const quoteItemCount = quoteMock.items?.[0]?.items?.length
+
+  await waitFor(() => {
+    expect(screen.queryByTestId('quoteItems-length')).toHaveTextContent(quoteItemCount.toString())
+  })
+
+  server.use(
+    graphql.query('getQuoteByID', (_req, res, ctx) => {
+      return res(
+        ctx.data({
+          ...singleQuoteItemMock,
+          quote: {
+            items: [
+              ...(singleQuoteItemMock.quote?.items as CrOrderItem[]),
+              ...quoteMock?.items?.[1]?.items,
+            ],
+          },
+        })
+      )
+    })
+  )
+
+  user.click(screen.getByTestId('add-non-configurable-product-button'))
+
+  await waitFor(() => {
+    expect(screen.queryByTestId('quoteItems-length')).toHaveTextContent(
+      (quoteItemCount + 1).toString()
+    )
+  })
+}
+
+afterEach(() => {
+  cleanup()
+})
+
+describe('[components] CreateNewQuoteTemplate', () => {
+  describe('Desktop', () => {
+    beforeEach(() => {
+      const useMediaQueryMock = useMediaQuery as jest.Mock
+      useMediaQueryMock.mockReturnValue(true)
+    })
+    it('should render CreateNewQuoteTemplate component', async () => {
+      renderWithQueryClient(<Common {...Common.args} />)
+
+      const quotesText = screen.getByText(/quotes/i)
+      const createQuote = screen.getByText(/create-quote/i)
+      const quoteDetails = screen.getByText(/quote-details/i)
+      const quoteSummary = screen.getByText(/quote-summary/i)
+
+      const clearChangesButton = screen.getByRole('button', { name: 'clear-changes' })
+      const saveAndExitButton = screen.getByRole('button', { name: 'save-and-exit' })
+      const submitForApprovalButton = screen.getByRole('button', { name: 'submit-for-approval' })
+      const printQuoteButton = screen.getByRole('button', { name: 'print-quote' })
+      const b2bProductComponent = screen.getByTestId('b2b-product-search-component')
+      //   const promoCodeComponent = screen.getByTestId('promo-code-badge-component')
+      //   const orderTotalComponent = screen.getByTestId('key-value-display-component')
+
+      expect(quotesText).toBeVisible()
+      expect(createQuote).toBeVisible()
+      expect(quoteDetails).toBeVisible()
+      expect(quoteSummary).toBeVisible()
+      expect(clearChangesButton).toBeVisible()
+      expect(saveAndExitButton).toBeVisible()
+      expect(submitForApprovalButton).toBeVisible()
+      expect(printQuoteButton).toBeVisible()
+      expect(b2bProductComponent).toBeVisible()
+
+      await waitFor(() => {
+        const quickOrderTableComponent = screen.queryByTestId('b2b-product-details-table-component')
+        expect(quickOrderTableComponent).toBeVisible()
+      })
+
+      //   expect(promoCodeComponent).toBeVisible()
+      //   expect(orderTotalComponent).toBeVisible()
+    })
+
+    xit('should add product in list when user clicks on non configurable product', async () => {
```

Comment:
what is xit?
clean up the unnecessary commented code

Reply:
will handle in another ticket(view and edit quote).